### PR TITLE
Fixing DAG structure and return value for Spark service cluster (resolves #162)

### LINKED ISSUE #162: Pickling fails in adam_gatk_pipeline
Some issue with a promise not getting picked properly, tbd.

```
WARNING:toil.leader:9a29c6ce-7fc8-42b8-8c14-faaa2032e4c5:         File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1200, in _serialiseJob
WARNING:toil.leader:9a29c6ce-7fc8-42b8-8c14-faaa2032e4c5:       PicklingError: Can't pickle <type 'instancemethod'>: attribute lookup __builtin__.instancemethod failed
```

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #162.

Doesn't work yet; I'll be offline most of the day tomorrow, but just wanted to push this so that people know what changes I'm making, since we've got lots of things in flight.

### CODE REVIEW

**Comment by fnothaft** [comment]
This is good to go. Ping @hannes-ucsc @almussel for review/merge.

File: src/toil_scripts/spark_utils/spawn_cluster.py [hunk 1]
```diff
@@ -6,44 +6,44 @@
...
+import multiprocessing
+import os
+from subprocess import call, check_call, check_output
+import time
 
 from toil.job import Job
 from toil_scripts.batch_alignment.bwa_alignment import docker_call
 
-log = logging.getLogger(__name__)
+_log = logging.getLogger(__name__)
```
Comment (heuermh) [inline src/toil_scripts/spark_utils/spawn_cluster.py]: Why do we have to this?  I probably don't want to know . . .

Comment (fnothaft) [inline src/toil_scripts/spark_utils/spawn_cluster.py]: It's related to @hannes-ucsc's earlier comment RE: not `from <module> import *`ing unless `__all__` is defined. TL;DR is that prefixing a field/method with `_` is a Python convention for something being private. It's more a convention than a hard access control, though.

File: src/toil_scripts/spark_utils/spawn_cluster.py [hunk 2]
```diff
@@ -6,44 +6,44 @@
...
-Master service job for a Spark/HDFS cluster.
-
-Starts up a Spark master and HDFS namenode. Return value of the service
-job is the instance IP.
-
-@author Audrey Musselman-Brown, almussel@ucsc.edu
-@author Frank Austin Nothaft, fnothaft@berkeley.edu
-"""
-
+    job.addChildJobFn(jFn, masterIP, *jArgs)
```
Comment (fnothaft) [inline src/toil_scripts/spark_utils/spawn_cluster.py]: Need to def'n cores/mem/disk.

**Pull request merged**